spec/boxcars/zero_shot_spec.rb

```diff
@@ -4,22 +4,24 @@
   let(:engine) { Boxcars::Openai.new }
   let(:search) { Boxcars::GoogleSearch.new }
   let(:calculator) { Boxcars::Calculator.new(engine: engine) }
-  let(:train) { described_class.new(engine: engine, boxcars: [search, calculator]) }
 
   it "can execute a train" do
+    train = described_class.new(engine: engine, boxcars: [search, calculator])
+
     VCR.use_cassette("zeroshot") do
       question = "What is pi times the square root of the average temperature in Austin TX in January?"
       expect(train.run(question)).to eq("25.132741228718345")
     end
   end
 
   context "with sample helpdesk app" do
-    helpdesk = Boxcars::ActiveRecord.new(models: [Comment, Ticket, User], name: "helpdesk")
-    new_train = described_class.new(boxcars: [helpdesk, Boxcars::Calculator.new, Boxcars::GoogleSearch.new])
+    let(:helpdesk) { Boxcars::ActiveRecord.new(models: [Comment, Ticket, User], name: "helpdesk") }
+
     it "can use a train to answer a question" do
+      train = described_class.new(boxcars: [helpdesk, calculator, search])
```

Comment:
I reused variables here. The only difference is in the calculator, but tests passed fine.